File: apps/ios/PagApp/PagApp/Views/Home/Story/PAGStoryBar.swift
import SwiftUI

public struct PAGStoryBar: View {
    public let items: [StoryItemType]
    public let onSelect: (StoryItemType) -> Void
    
    public init(items: [StoryItemType], onSelect: @escaping (StoryItemType) -> Void) {
        self.items = items
        self.onSelect = onSelect
    }
    
    public var body: some View {
        ScrollView(.horizontal, showsIndicators: false) {
            HStack(spacing: PAGSpacing.sm) {
                ForEach(items, id: \.self) { item in
                    PAGStoryItem(type: item) {
                        onSelect(item)
                    }
                }
            }
            .padding(.horizontal, PAGSpacing.sm)
        }
        .padding(.vertical, PAGSpacing.sm)
        .background(PAGTheme.backgroundPrimary)
        .overlay(
            Rectangle()
                .frame(height: 1)
                .foregroundColor(PAGTheme.borderDefault),
            alignment: .bottom
        )
    }
}

File: apps/android/app/src/main/java/com/alafteknoloji/pagapp/ui/screens/home/story/PAGStoryBar.kt
package com.alafteknoloji.pagapp.ui.screens.home.story

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.Divider
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.alafteknoloji.pagapp.models.StoryItemType
import com.alafteknoloji.pagapp.ui.theme.PAGTheme

@Composable
fun PAGStoryBar(
    items: List<StoryItemType>,
    onSelect: (StoryItemType) -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(PAGTheme.colors.backgroundPrimary)
    ) {
        LazyRow(
            modifier = Modifier.fillMaxWidth(),
            contentPadding = PaddingValues(horizontal = PAGTheme.spacing.sm),
            horizontalArrangement = Arrangement.spacedBy(PAGTheme.spacing.sm)
        ) {
            items(items) { item ->
                PAGStoryItem(
                    item = item,
                    onClick = { onSelect(item) }
                )
            }
        }
        
        Spacer(modifier = Modifier.height(PAGTheme.spacing.sm))
        
        Divider(
            color = PAGTheme.colors.borderDefault,
            thickness = 1.dp
        )
    }
}
